addOns/oast/src/main/java/org/zaproxy/addon/oast/OastInsertExistingPayload.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.oast;
+
+import javax.swing.JMenu;
+import javax.swing.JMenuItem;
+import org.parosproxy.paros.Constant;
+
+@SuppressWarnings("serial")
+public class OastInsertExistingPayload extends JMenu {
+    private ExtensionOast extension = null;
+    private JMenuItem boastMenuItem = null;
+    private JMenuItem interactshMenuItem = null;
+    private JMenuItem callbackMenuItem = null;
+    private OastInsertPayloadMenu parentMenu = null;
+
+    public OastInsertExistingPayload(OastInsertPayloadMenu parentMenu, ExtensionOast extension) {
+        super(Constant.messages.getString("oast.popup.menu.existing"));
+        this.parentMenu = parentMenu;
+        this.extension = extension;
+        initialize();
+    }
+
+    public void initialize() {
+        boastMenuItem = new JMenuItem(Constant.messages.getString("oast.popup.menu.boast"));
+        interactshMenuItem =
+                new JMenuItem(Constant.messages.getString("oast.popup.menu.interactsh"));
+        callbackMenuItem = new JMenuItem(Constant.messages.getString("oast.popup.menu.callback"));
+        this.add(boastMenuItem);
+        this.add(interactshMenuItem);
+        this.add(callbackMenuItem);
+        addBoastActionListener();
+        addInteractshActionListener();
+        addCallbackActionListener();
+    }
+
+    public void addBoastActionListener() {
+        boastMenuItem.addActionListener(
+                new java.awt.event.ActionListener() {
+                    @Override
+                    public void actionPerformed(java.awt.event.ActionEvent e) {
```

Comment:
All three of these methods could use lambdas for the listener, to shorten the code a bit.